pkg/scalers/splunk_observability_scaler.go

```diff
@@ -0,0 +1,183 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"regexp"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/signalfx/signalflow-client-go/v2/signalflow"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type splunkObservabilityMetadata struct {
+	TriggerIndex int
+
+	AccessToken           string  `keda:"name=accessToken,          order=authParams"`
+	Realm                 string  `keda:"name=realm,                order=authParams"`
+	Query                 string  `keda:"name=query,                order=triggerMetadata"`
+	Duration              int     `keda:"name=duration,             order=triggerMetadata"`
+	TargetValue           float64 `keda:"name=targetValue,   	   order=triggerMetadata"`
+	QueryAggregator       string  `keda:"name=queryAggregator,      order=triggerMetadata"`
+	ActivationTargetValue float64 `keda:"name=activationTargetValue, order=triggerMetadata"`
+}
+
+type splunkObservabilityScaler struct {
+	metadata  *splunkObservabilityMetadata
+	apiClient *signalflow.Client
+	logger    logr.Logger
+}
+
+func parseSplunkObservabilityMetadata(config *scalersconfig.ScalerConfig) (*splunkObservabilityMetadata, error) {
+	meta := &splunkObservabilityMetadata{}
+	meta.TriggerIndex = config.TriggerIndex
+
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing splunk observability metadata: %w", err)
+	}
+
+	return meta, nil
+}
+
+func newSplunkO11yConnection(meta *splunkObservabilityMetadata, logger logr.Logger) (*signalflow.Client, error) {
+	logger.Info(fmt.Sprintf("meta: %+v\n", meta))
+
+	if meta.Realm == "" || meta.AccessToken == "" {
+		return nil, fmt.Errorf("error: Could not find splunk access token or ream")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(meta.Realm),
+		signalflow.AccessToken(meta.AccessToken),
+		signalflow.OnError(func(err error) {
+			errorMsg := fmt.Sprintf("error in SignalFlow client: %v\n", err)
```

Comment:
To concatenate the error message, I think it would be better to use the error string like so.

`errorMsg := fmt.Sprintf("error in SignalFlow client: %s\n", err.Error())`

Since it's an error, we should use `logger.Error()` instead of `logger.Info()`